### PR TITLE
ci: Capping the dask dependency since Dask dropped Python 3.7 support

### PR DESCRIPTION
**Pull request by adchia**
**What this PR does / why we need it**:
See https://github.com/dask/dask/pull/8572 which dropped Python 3.7 support in Dask for the last 2 releases.

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: sdk/python/setup.py [hunk 1]
```diff
@@ -40,7 +40,7 @@
 REQUIRES_PYTHON = ">=3.7.0"
 
 REQUIRED = [
-    "Click==8.*",
+    "Click>=7.*",
```
Comment (achals) [inline sdk/python/setup.py:43]: Was this necessary for the same reason (python 3.7 support)?

Comment (adchia) [reply sdk/python/setup.py:43]: ah no this was just a fix i wanted to sneak in from a previous PR because it seemed overly restrictive

Comment (achals) [reply sdk/python/setup.py:43]: sg. stamping.

**Review by achals — APPROVED** [review]
/lgtm

File: sdk/python/setup.py (hunk 1 above)
Comment (tpvasconcelos) [reply sdk/python/setup.py:43]: @adchia @achals I think this was a typo as pip actually interprets this as `click==7.*`. See https://github.com/feast-dev/feast/pull/2450

**Pull request merged**